source/TradingPanel.cpp

```diff
@@ -337,3 +336,79 @@ void TradingPanel::Buy(int64_t amount)
 	player.Accounts().AddCredits(-amount * price);
 	GameData::AddPurchase(system, type, amount);
 }
+
+
+
+void TradingPanel::SellOutfitsOrMinables(bool sellMinable)
+{
+	int day = player.GetDate().DaysSinceEpoch();
+	for(const auto &it : player.Cargo().Outfits())
+	{
+		if(sellMinable != static_cast<bool>(it.first->Get("minable")))
+			continue;
+		int64_t value = player.FleetDepreciation().Value(it.first, day, it.second);
+		profit += value;
+		tonsSold += static_cast<int>(it.second * it.first->Mass());
+
+		player.AddStock(it.first, it.second);
+		player.Accounts().AddCredits(value);
+		player.Cargo().Remove(it.first, it.second);
+	}
+}
+
+
+
+string TradingPanel::OutfitSalesMessage(bool sellMinable, size_t displayLimit)
+{
+	struct OutfitInfo
+	{
+		std::string name;
+		int64_t count, value;
+	};
+	vector<OutfitInfo> outfitValue;
+	double tonsSold = 0;
+	int profit = 0;
+	int day = player.GetDate().DaysSinceEpoch();
+	for(auto &it : player.Cargo().Outfits())
+	{
+		if(sellMinable != static_cast<bool>(it.first->Get("minable")))
+			continue;
+		if(!it.second)
+			continue;
+		int64_t value = player.FleetDepreciation().Value(it.first, day, it.second);
+		profit += value;
+		tonsSold += static_cast<int>(it.second * it.first->Mass());
+		// Store a description of the count & item, followed by its value.
+		outfitValue.push_back({ "", it.second, value });
```

Comment:
```suggestion
		outfitValue.emplace_back("", it.second, value);
```
this should work as we use C++20 now

Reply:
The mac and windows compilers don't like that change.

Reply:
According to the [implementation matrix](https://en.cppreference.com/w/cpp/20) ("Parenthesized initialization of aggregates"), Windows _should_ support this, but yeah, looks like Apple doesn't like it.